### ISSUE TITLE
Segment template regression from 2.5.x

### ISSUE THREAD

**Issue by tigrato** [google/shaka-player#2659]
**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**
Yes

**What version of Shaka Player are you using?**
3.0.0

**Can you reproduce the issue with our latest release version?**
yes

**Can you reproduce the issue with the latest code from `master`?**
yes

**Are you using the demo app or your own custom app?**
demo

**What browser and OS are you using?**
Chrome with macOS

**What are the manifest and license server URIs?**
```
<MPD xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="urn:mpeg:dash:schema:mpd:2011" xsi:schemaLocation="urn:mpeg:dash:schema:mpd:2011 http://standards.iso.org/ittf/PubliclyAvailableStandards/MPEG-DASH_schema_files/DASH-MPD.xsd" id="201" profiles="urn:mpeg:dash:profile:isoff-live:2011" type="dynamic" availabilityStartTime="2020-05-09T19:33:40Z" publishTime="2020-06-17T12:32:13Z" minimumUpdatePeriod="PT10S" minBufferTime="PT30S" suggestedPresentationDelay="PT12S">
  <Period id="616164" start="PT928H56M10.656S" duration="PT1M59.960S">
    <AdaptationSet mimeType="video/mp4" startWithSAP="1" segmentAlignment="true" bitstreamSwitching="true" subsegmentAlignment="true" subsegmentStartsWithSAP="1">
      <Representation width="640" height="360" frameRate="25/1" codecs="avc1.42C01E" id="1" bandwidth="1000000">
        <SegmentTemplate timescale="25000" presentationTimeOffset="83604266400" startNumber="616165" media="index_video_9_0_$Time$.mp4?m=1583334383" initialization="index_video_9_0_init.mp4?m=1583334383">
          <SegmentTimeline>
            <S t="83604266400" d="75000"/>
            <S t="83604341400" d="150000" r="18"/>
            <S t="83607191400" d="74000"/>
          </SegmentTimeline>
        </SegmentTemplate>
      </Representation>
      <Representation width="960" height="540" frameRate="25/1" codecs="avc1.42C01F" id="2" bandwidth="2000000">
        <SegmentTemplate timescale="25000" presentationTimeOffset="83604266400" startNumber="616165" media="index_video_10_0_$Time$.mp4?m=1583334383" initialization="index_video_10_0_init.mp4?m=1583334383">
          <SegmentTimeline>
            <S t="83604266400" d="75000"/>
            <S t="83604341400" d="150000" r="18"/>
            <S t="83607191400" d="74000"/>
          </SegmentTimeline>
        </SegmentTemplate>
      </Representation>
      <Representation width="1280" height="720" frameRate="25/1" codecs="avc1.4D401F" id="3" bandwidth="3000000">
        <SegmentTemplate timescale="25000" presentationTimeOffset="83604266400" startNumber="616165" media="index_video_11_0_$Time$.mp4?m=1583334383" initialization="index_video_11_0_init.mp4?m=1583334383">
          <SegmentTimeline>
            <S t="83604266400" d="75000"/>
            <S t="83604341400" d="150000" r="18"/>
            <S t="83607191400" d="74000"/>
          </SegmentTimeline>
        </SegmentTemplate>
      </Representation>
      <Representation width="1920" height="1080" frameRate="25/1" codecs="avc1.4D4028" id="4" bandwidth="5000000">
        <SegmentTemplate timescale="25000" presentationTimeOffset="83604266400" startNumber="616165" media="index_video_12_0_$Time$.mp4?m=1583334383" initialization="index_video_12_0_init.mp4?m=1583334383">
          <SegmentTimeline>
            <S t="83604266400" d="75000"/>
            <S t="83604341400" d="150000" r="18"/>
            <S t="83607191400" d="74000"/>
          </SegmentTimeline>
        </SegmentTemplate>
      </Representation>
    </AdaptationSet>
    <AdaptationSet mimeType="audio/mp4" lang="eng">
      <Representation audioSamplingRate="48000" codecs="mp4a.40.2" id="5" bandwidth="96348">
        <AudioChannelConfiguration schemeIdUri="urn:mpeg:dash:23003:3:audio_channel_configuration:2011" value="2"/>
        <SegmentTemplate timescale="48000" presentationTimeOffset="160520191552" startNumber="616165" media="index_audio_13_0_$Time$.mp4?m=1583334383" initialization="index_audio_13_0_init.mp4?m=1583334383">
          <SegmentTimeline>
            <S t="160520191552" d="144384"/>
            <S t="160520335936" d="287744"/>
            <S t="160520623680" d="288768"/>
            <S t="160520912448" d="287744" r="2"/>
            <S t="160521775680" d="288768"/>
            <S t="160522064448" d="287744" r="2"/>
            <S t="160522927680" d="288768"/>
            <S t="160523216448" d="287744" r="2"/>
            <S t="160524079680" d="288768"/>
            <S t="160524368448" d="287744" r="2"/>
            <S t="160525231680" d="288768"/>
            <S t="160525520448" d="287744"/>
            <S t="160525808192" d="142336"/>
          </SegmentTimeline>
        </SegmentTemplate>
      </Representation>
    </AdaptationSet>
  </Period>
  <Period id="PID_0_5ec02fa8dd7" start="PT928H58M10.616S" duration="PT20S">
    <AdaptationSet frameRate="25/1" mimeType="video/mp4" startWithSAP="1" segmentAlignment="true" subsegmentAlignment="true" subsegmentStartsWithSAP="1">
      <SegmentTemplate timescale="90000">
        <SegmentTimeline>
          <S d="540000" r="2"/>
          <S t="1620000" d="180000"/>
        </SegmentTimeline>
      </SegmentTemplate>
      <Representation width="1920" height="1080" codecs="avc1.4d4028" scanType="progressive" id="1" bandwidth="4000000">
        <SegmentTemplate media="index_FHD_$Time$.mp4" initialization="index_FHD_init.mp4">
        </SegmentTemplate>
      </Representation>
      <Representation width="960" height="540" codecs="avc1.42c01f" scanType="progressive" id="2" bandwidth="2000000">
        <SegmentTemplate media="index_SD_$Time$.mp4" initialization="index_SD_init.mp4">
        </SegmentTemplate>
      </Representation>
      <Representation width="1280" height="720" codecs="avc1.4d401f" scanType="progressive" id="3" bandwidth="2000000">
        <SegmentTemplate media="index_HD_$Time$.mp4" initialization="index_HD_init.mp4">
        </SegmentTemplate>
      </Representation>
      <Representation width="640" height="360" codecs="avc1.42c01e" scanType="progressive" id="4" bandwidth="1000000">
        <SegmentTemplate media="index_WDF_$Time$.mp4" initialization="index_WDF_init.mp4">
        </SegmentTemplate>
      </Representation>
    </AdaptationSet>
    <AdaptationSet mimeType="audio/mp4" lang="und">
      <SegmentTemplate timescale="48000" media="index_audio_$Time$.mp4" initialization="index_audio_init.mp4">
        <SegmentTimeline>
          <S d="289792"/>
          <S t="289792" d="288768"/>
          <S t="578560" d="287744"/>
          <S t="866304" d="94208"/>
        </SegmentTimeline>
      </SegmentTemplate>
      <Representation audioSamplingRate="48000" codecs="mp4a.40.2" id="5" bandwidth="96000">
        <AudioChannelConfiguration schemeIdUri="urn:mpeg:dash:23003:3:audio_channel_configuration:2011" value="2"/>
      </Representation>
    </AdaptationSet>
  </Period>
</MPD>
```

**What did you do?**
I tried to play a multi-period stream where each period has a different segment template with the latest shaka version. Previous versions are working just fine.

**What did you expect to happen?**
Once the player reaches the period `PID_0_5ec02fa8dd7` it should load the `SegmentTemplate` from the representation and use the `SegmentTimeline` from adaptation.

**What actually happened?**
When the player reaches the period `PID_0_5ec02fa8dd7` it uses the `SegmentTemplate` defined in the previous period with the BaseURL from the current period. 

It seems that the shaka is not reloading `SegmentTemplate`  at least for this case where the timeline is defined in a different place than the  `SegmentTemplate`

**Comment by tigrato** [comment]
its a duplicate of #2650

**Comment by tigrato** [comment]
Might be fixed already by #2650

**Comment by joeyparrish** [comment]
Yes, this is a duplicate of #2650, but that is not clear from the title of that issue.  I'll update #2650's title to make it clear what the problem turned out to be.

@tigrato, if you try the latest code from `master` or our [nightly demo](https://nightly-dot-shaka-player-demo.appspot.com/demo/), this should be working now.  We will be releasing v3.0.1 soon with this fix included.

**Comment by joeyparrish** [comment]
Please let us know if it's working or not, and we can reopen this issue if needed.  Thanks!

**Comment by tigrato** [comment]
@joeyparrish it's ok in nightly. 

I have tested it before but I was afraid that it wasn't the master since in the top it has: v2.5.13-master-11-g810f09ff and this bug as v3.0.0 and none of the previous versions have the same issue.

Thanks!

**Comment by joeyparrish** [comment]
Sorry about that.  The master tags are a bit confusing.

v2.5.13-master-11-... means that we've made 11 changes in master since v2.5.13 was released, even though v2.5.13 was released from the v2.5.x branch, not master.  The same will be true when we release v3.0.1 soon, even though it will be released from the v3.0.x branch (which is extremely close to master, since we only branched a week or so ago).
